packages/perseus/src/components/math-input.tsx

```diff
@@ -258,36 +270,201 @@ class MathInput extends React.Component<Props, State> {
-            // @ts-expect-error - TS2322 - Type 'Element' is not assignable to type 'null'.
-            buttons = (
-                <TexButtons
-                    sets={this.props.buttonSets}
-                    className="math-input-buttons absolute"
-                    convertDotToTimes={this.props.convertDotToTimes}
-                    onInsert={this.insert}
-                />
-            );
-        }
-
         return (
-            <div style={{display: "inline-block"}}>
-                <div style={{display: "inline-block"}}>
+            <View
+                style={[
+                    styles.outerWrapper,
+                    this.state.focused && styles.wrapperFocused,
+                    this.props.hasError && styles.wrapperError,
+                ]}
+            >
+                <div
+                    style={{
+                        display: "flex",
+                        padding: 1,
+                    }}
+                    onClick={() => {
+                        const mathField = this.mathField();
+                        if (!mathField) {
+                            return;
+                        }
+                        this.setState({
+                            cursorContext: getCursorContext(mathField),
+                        });
+                    }}
+                >
                     <span
                         className={className}
                         ref={(ref) => (this.__mathFieldWrapperRef = ref)}
                         aria-label={this.props.labelText}
-                        onFocus={this.handleFocus}
-                        onBlur={this.handleBlur}
+                        onFocus={() => this.focus()}
+                        onBlur={() => this.blur()}
                     />
+                    <Popover
+                        opened={this.state.keypadOpen}
+                        onClose={() => this.closeKeypad()}
+                        content={() => (
+                            <PopoverContentCore
+                                closeButtonVisible
+                                style={styles.popoverContent}
+                            >
+                                <DesktopKeypad
+                                    onAnalyticsEvent={
+                                        this.props.analytics.onAnalyticsEvent
+                                    }
+                                    extraKeys={this.props.extraKeys}
+                                    onClickKey={this.handleKeypadPress}
+                                    cursorContext={this.state.cursorContext}
+                                    convertDotToTimes={
+                                        this.props.convertDotToTimes
+                                    }
+                                    {...(this.props.keypadButtonSets ??
+                                        mapButtonSets(this.props?.buttonSets))}
+                                />
```

Comment:
**question**: I was doing some work in webapp related to the use of the mobile keypad and got thinking. Is there a way we can have both keypads rendered in the same place?  Today, if we land this the mobile keypad will be rendered/controlled by the host (webapp) but we'll render the desktop keypad here.  I think we're improving the keypad situation with these PRs, but I wonder if we can unify where we render the different keypads so there's a single place/component that controls that. 

See https://github.com/Khan/webapp/blob/master/services/static/javascript/content-library/ui/content-renderers/exercise/exercise-content-footer.tsx#L305 and https://github.com/Khan/webapp/blob/12197234a8fcca280fd896c15b44bfa8fe800438/services/static/javascript/content-library/ui/content-footer.tsx#L168.  

Maybe also something to work with @handeyeco on?